gradio/flagging.py

```diff
@@ -548,3 +548,17 @@ def get_unique_name(self):
     def dump_json(self, thing: dict, file_path: str) -> None:
         with open(file_path, "w+", encoding="utf8") as f:
             json.dump(thing, f)
+
+
+class FlagMethod:
+    """
+    Helper class that contains the flagging button option and callback
+    """
+
+    def __init__(self, flagging_callback, flag_option=None):
+        self.flagging_callback = flagging_callback
+        self.flag_option = flag_option
+        self.__name__ = "Flag"
```

Comment:
why set `__name__`?

Reply:
This existed in the `FlagMethod` class already, I didn't add it. However, I think the reason for it is because in Blocks, the event listener needs a name for the function